### DISCUSSION TITLE
Spot Lights not casting shadows on ground plane

### DISCUSSION THREAD

**Discussion by RonMehta** [google/filament#6179] [Q&A]
Hi

I am trying to cast shadows on the ground plane via spotlight, though the shadows are being casted from one object to another they are not casted on a ground plane like it does for directional light.

I tried to add the code in shadowtest.cpp to test

`	LightManager::Builder(LightManager::Type::POINT)
				.color(Color::toLinear<ACCURATE>(sRGBColor(1.0f, 0.0f, 0.0f)))
				.intensity(2000000.0f, LightManager::EFFICIENCY_LED)
				.position({ 5, 5, 5})
				.direction({   -5, -5, -5 })
				.spotLightCone(static_cast<float>(M_PI / 60), static_cast<float>((M_PI / 60)))
				.falloff(50.0f)
				.castShadows(true)
				.build(*engine, app.lights.back());
		scene->addEntity(app.light);`

Is this not possible? Can anyone help me with this
Thanks in advance

**Comment by RonMehta** [comment]
@pixelflinger can you comment on this?

**Comment by romainguy** [comment]
Is this an invisible ground plane? If yes, it's not supported at the moment

**Reply by RonMehta** [reply]
thanks for the reply @romainguy, can you point me a way to do it?

**Reply by RonMehta** [reply]
Hi @romainguy can you guide me on this, what would be the required changes we might need to make to support shadows on the invisible ground and what might the challenges be with this kind of requirement?
